winCLI-EN.py

```diff
@@ -0,0 +1,274 @@
+                    click.echo(f"    Edited Save: {is_edited}")
+                    click.echo(f"    Creative Mode Enabled: {is_creative_mode}")
+            
+            if idx == current_session_index:
+                click.echo(f"  * This is the current active session.")
+
+        click.echo("\nEnumeration completed.")
+
+
+
+def read_config():
+    config = configparser.ConfigParser()
+    config.read(CONFIGFILE)
+    if "server" not in config.sections():
+        config.add_section("server")
+
+    if "token" not in config.options("server"):
+        config['server'] = {"token": ""}
+
+    return config
+
+
+def save_config(config):
+    with open(CONFIGFILE, "w+") as f:
+        config.write(f)
+
+
+def send_command(token, funcName, data=None):
+    try:
+        headers = {}
+        if token:
+            headers = {
+                'Authorization': f'Bearer {token}',
+                'Content-Type': 'application/json'
+            }
+
+        jsonreq = {
+            "function": funcName
+        }
+
+        if data:
+            jsonreq["data"] = data
+
+        response = requests.post(SERVER_URL, headers=headers, verify=False, json=jsonreq)
+
+        if response.status_code >= 200 and response.status_code < 300:
+            click.echo(f"Command success: {response.status_code}")
+            return response
+        else:
+            click.echo(f"Failed to execute command: {response.status_code} {response.reason}")
+            click.echo(response.text)
+            return None
+    except requests.exceptions.RequestException as e:
+        click.echo(f"An error occurred: {e}")
+        raise e
+
+
+def display_menu():
+    """Display the menu for selecting a command."""
+    click.echo("\nChoose a command:")
+    click.echo("1. Show server status")
+    click.echo("2. Save game")
+    click.echo("3. Shutdown server")
+    click.echo("4. Enumerate sessions")
+    click.echo("5. Show server options")  # New option
+    click.echo("6. Exit\n")  # Updated option number
+
+
+@click.command()
+@click.option('--host', 'host', default="localhost:7777", help='Host:port to connect to')
+@click.option('--password', hide_input=True, help='Password for server authentication.')
+def cli(host, password):
+    """CLI tool to authenticate and interact with the Satisfactory Dedicated Server API."""
+    config = read_config()
+    token = config.get("server", "token")
+
+    if not token:
+        # If token is absent, prompt for the password
+        password = click.prompt("Password", hide_input=True)
+
+        click.echo("Authenticating...")
+        token = authenticate(password)
+        if token:
+            config['server']['token'] = token
+            save_config(config)
+        else:
+            click.echo("Authentication failed.")
+            return
+
+    if host:
+        global SERVER_URL
+        SERVER_URL = f'https://{host}/api/v1'
+
+    # Menu after authentication
+    while True:
+
+        display_menu()
```

Comment:
this menu kinda makes it not a cli anymore, it's a UI that happen to be text based. not something i personally care for, but since it doesnt change the main cli.py, that's fine for now. 

we might consider merging the two and having a --interactive flag instead that starts the menu

Reply:
Now it was GUI + CLI